staging/vhost-device-gpu/src/virtio_gpu.rs

```diff
@@ -764,6 +777,8 @@ impl VirtioGpu for RutabagaVirtioGpu {
             );
             ErrUnspec
         })?;
+
+        self.shared_uuid.insert(resource_id, shared_msg);
```

Comment:
What if the resource is already exported, though? (So it is in the `shared_uuid` map). The spec doesn't say anything about this, but I think it's worth to think about it. I think we could just return the already assigned uuid*. What do you think?

Btw I think there should be a more efficient way to do that using only 1 lookup using the [Entry API](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.entry), instead of doing separate `get` and `insert`.

Reply:
the QEMU code for `add_vhost_device` says 
```
 * Return: true if the UUID did not exist and the device has been tracked,
 * false if another resource with the same UUID already existed.
 * Note that if it finds a repeated UUID, the resource is not inserted in
 * the lookup table.
 ```
so i think we could just retrieve and return the existing UUID without inserting a new entry like so
 ```
    if let Some(existing_resource) = self.shared_resources.get(&resource_id) {
        return Ok(OkResourceUuid {
            uuid: existing_msg.uuid.into_bytes(),
        });
    }
```
 or do nothing about it, since in qemu the resource is not inserted anyway.